### ISSUE TITLE
FreeBSD accf_data accept() filter support

### ISSUE THREAD

**Issue by Tintinabula** [h2o/h2o#839]
Feature Request: FreeBSD accf_data accept() filter support

The accept() filter in FreeBSD will buffer incoming connections until client
data arrives at which point the daemon, like an https web server, will receive
the connected descriptor via accept(). If the client never sends any data then
the https daemon will not waist any processing time on the connection. The
filter should allow the H2O daemon to serve more clients by filtering out non
responsive clients.

FreeBSD Man Page, ACCF_DATA
https://www.freebsd.org/cgi/man.cgi?query=accf_data

The accf_data kernel module loaded by adding accf_data_load="YES" to
/boot/loader.conf

The accf_data and accf_http FreeBSD filters are currently supported in Apache,
Nginx and Varnish. I am not sure if the addition of the accf_data filter will
benefit H2O.

Linux has a similar, but incompatible accept filter:

TCP_DEFER_ACCEPT (since Linux 2.4) Allow a listener to be awakened only when
data arrives on the socket. Takes an integer value (seconds), this can bound
the maximum number of attempts TCP will make to complete the connection.

**Comment by kazuho** [comment]
Good catch!  H2O already supports TCP_DEFER_ACCEPT on Linux, but not ACCF_DATA for FreeBSD.
